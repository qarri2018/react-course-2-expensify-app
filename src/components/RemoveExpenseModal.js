import React from 'react';
import Modal from 'react-modal';

const RemoveExpenseModal = (props) => (

    <Modal
        className="modal"
        contentLabel="Remove Confirmation"
        onRequestClose={props.onRemove}
        isOpen={props.showModal}
        closeTimeoutMS={200}
    >
        <h3 className="modal-title">Are you sure?</h3>
        <div className="modal__body">
            <button className="button remove-expense-modal__button-yes" onClick={props.onRemove}>Yes</button>
            <button className="button button--secondary remove-expense-modal__button-no" onClick={props.hideModal}>No</button>
        </div>


    </Modal>

);

export default RemoveExpenseModal;