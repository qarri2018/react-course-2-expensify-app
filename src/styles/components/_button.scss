.button {
    background: $blue;
    color: white;
    display: inline-block;
    border: none;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
}