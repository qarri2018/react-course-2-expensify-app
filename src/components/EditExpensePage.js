import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import RemoveExpenseModal from './RemoveExpenseModal';


export class EditExpensePage extends React.Component {
  state = { showModal: false };

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  showModal = () => {
    this.setState({ showModal: true });
  };    
  hideModal = () => {
    this.setState({ showModal: false });
  };  
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.showModal}>Remove Expense</button>
        </div>
        <RemoveExpenseModal
          showModal={this.state.showModal}
          hideModal={this.hideModal}
          onRemove={this.onRemove}
        />        
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
