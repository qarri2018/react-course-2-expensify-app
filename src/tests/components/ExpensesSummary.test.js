import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

// Should render the component with one expense
test('should render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={2354848139847} />);
    expect(wrapper).toMatchSnapshot();
});

