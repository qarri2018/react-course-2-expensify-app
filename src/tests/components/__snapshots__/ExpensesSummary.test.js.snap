// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
     Viewing 
    1
     
    expense
     totalling 
    $2.35
      
  </h1>
</div>
`;

exports[`should render ExpensesSummary with one expense 2`] = `
<div>
  <h1>
     Viewing 
    23
     
    expenses
     totalling 
    $23,548,481,398.47
      
  </h1>
</div>
`;
